nx/lib/nx/binary_backend/matrix.ex

```diff
@@ -860,7 +867,7 @@ defmodule Nx.BinaryBackend.Matrix do
       |> Enum.map(fn col ->
         row
         |> Enum.zip(col)
-        |> Enum.reduce(0, fn {x, y}, acc -> acc + x * y end)
+        |> Enum.reduce(0, fn {x, y}, acc -> acc + Complex.multiply(x, y) end)
```

Comment:
This is wrong. It must be one times the complex conjugate of the other.

By the way, this module imports Complex.Kernel, so the * operator is the same as Complex.multiply

Reply:
I saw that fixing this had broken a test, but the definition must be so, and one way to think about that is that a dot(v, v) must yield norm(v) squared, and that will only happen if you conjugate one of the operands

Reply:
> This is wrong. It must be one times the complex conjugate of the other.

In `dot_matrix`, I just want to do a simple matrix product (A*B), not A*B^*. Why do I have to take the complex conjugate here?  By the way, if I take the complex conjugate here, I get an error in the test.

> By the way, this module imports Complex.Kernel, so the * operator is the same as Complex.multiply

Thank you for letting us know!

Reply:
The dot product for complex vectors is defined as A^*.B instead of A^t.B. The thing here is that it's the adjoint vector that matters, not the transpose in itself, and the adjoint for a complex vector is the transpose conjugate.

Rule of thumb is that every time you see a transpose in real matrices you swap it with a transpose conjugate for complex extension. 

One of the reasons is what I mentioned about dot(v, v) === norm(v)^2

https://en.m.wikipedia.org/wiki/Dot_product#Complex_vectors

Reply:
Thank you, I will fix the `dot_matrix` to that of suggestion and correct the incorrect use of `dot_matrix`.